### PR TITLE
Apply initial sim time also after a reset.

### PR DESCRIPTION
**Pull request by peci1**
# 🦟 Bug fix

Fixes https://github.com/gazebosim/gazebo-classic/pull/3294#issuecomment-1618394550

## Summary
Initial sim time should also be respected after a reset.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by traversaro — APPROVED** [review]
Thanks! Ok for me, as long as the ABI checker is happy (it should be, but better double check).

**Comment by peci1** [comment]
Thanks for the quick action! I just hope the other times present in `ResetTime()` are good to be zeros after reset. I tried to understand the meaning of each of them and it seemed so to me, but I'd rather have second eyes on this.

**Comment by traversaro** [comment]
Probably we can expand the setting description in https://github.com/gazebosim/gazebo-classic/blob/af7d3278a1cfae54328bfd2d4eba7b1c29ee669c/gazebo/Server.cc#L202 to explain that is not only used during the first run of the sim, but also after reset?

**Review by traversaro — APPROVED** [review]
Ok for me, thanks! Technically (per [Hyrum's Law](https://www.hyrumslaw.com/)) this may be a behaviour breaking change, but I think it make sense. Probably to be sure it is best to also wait for an approval of @scpeters .

**Comment by peci1** [comment]
> Technically (per [Hyrum's Law](https://www.hyrumslaw.com/)) this may be a behaviour breaking change

Technically, you're right. Practically, if anyone had tried the reset from e.g. current UNIX timestamp to 0, he'd face lots of problems with downstream code (e.g. ROS nodes). So my assumption is nobody tried that. The only potentially problematic use-case could be if the user specified a low init time like 1 or 10...

**Review by scpeters — APPROVED** [review]
this makes sense to me

thanks!

**Pull request merged**
